core/src/main/java/io/grpc/inprocess/InProcessServerBuilder.java

```diff
@@ -81,7 +81,18 @@
    * @return a new builder
    */
   public static InProcessServerBuilder forName(String name) {
-    return new InProcessServerBuilder(name);
+    return new InProcessServerBuilder(name, false);
+  }
+
+  /**
+   * Create a server builder for an anonymous in-process server.
+   * Anonymouns servers can only be connected to via their listen address,
+   * and can't be referenced by name.
+   * @param name a server identifier used for logging purposes only.
+   * @return a new builder
+   */
+  public static InProcessServerBuilder anonymous(String name) {
+    return new InProcessServerBuilder("anon:" + name, true);
```

Comment:
How necessary is this name? Could we just generate a random string for this?

Yes, it can show up in the logs. But in this model the server is more of just a "plain object" and POJO's don't have much of an identity. Maybe the debug name could be added later if 100% nameless became a problem?